src/bin/inx-chronicle/api/mod.rs

```diff
@@ -57,7 +58,10 @@ impl ApiWorker {
     }
 
     // TODO: What happens when there is an error in an API request. This should not bring down all of Chronicle.
-    pub async fn run(&self) -> Result<(), ApiError> {
+    pub async fn run<Fut>(&self, shutdown_handle: Fut) -> Result<(), ApiError>
```

Comment:
Maybe use `impl Future` and skip the `.then` part below?

Reply:
The `.then` is because the return type is specified in the graceful shutdown fn, so it's not avoidable. But we can use `impl Future`.